### ISSUE TITLE
Any plan to support readdir "withFileTypes" option?

### ISSUE THREAD

**Issue by zed-k** [tschaub/mock-fs#272]
[fast-glob](https://github.com/mrmlnc/fast-glob) uses this option since v3 in its ["modern mode"](https://github.com/nodelib/nodelib/blob/master/packages/fs/fs.scandir/README.md#old-and-modern-mode) to make fewer calls to file system.

Using it in combination with mock-fs leads to the following error:
```
Method not implemented

    at notImplemented (node_modules/mock-fs/lib/binding.js:140:9)
    at Object.<anonymous>.Binding.readdir (node_modules/mock-fs/lib/binding.js:890:5)
    at Object.<anonymous> (node_modules/mock-fs/lib/index.js:33:39)
    at readdirWithFileTypes (node_modules/@nodelib/fs.scandir/out/providers/async.js:15:17)
```

**Comment by tannerntannern** [comment]
I had to work around this in one of my projects, and in my case it was tolerable to simply disable "modern mode" by passing `stats: true` as an option to `fast-glob`, since modern mode is not used when the node version is <10.10, _or_ [stats is enabled](https://github.com/mrmlnc/fast-glob#old-and-modern-mode).

Another option would be to detect if mocha is running and conditionally disable modern mode, so _only your tests_ are affected by the performance hit, not the code your users run.  Something like this would help you determine if mocha is running:
```
// utils.js
export const IS_MOCHA = !!process.env.MOCHA_COLORS;
```

You could also try [detect-mocha](https://www.npmjs.com/package/detect-mocha), but I haven't used it myself.

**Comment by zed-k** [comment]
Thanks for the heads up, that's what I ended up doing.
It's not ideal but it works :)

**Comment by iOiurson** [comment]
I'm not sure to understand the workaround...
Is `fast-glob` a dependency of `fs` ?

Then how do you pass `stats: true` to `fast-glob` if you don't have it installed as direct dependency in your project ?

**Comment by zed-k** [comment]
It's indeed only a workaround for usage with `fast-glob`.

If your code (or one of your dependencies) make use of the new Node 10.10+ `fs.Dirent` class, you'll face the same error that I got, since `mock-fs` doesn't support it yet.

**Comment by mrmlnc** [comment]
@tschaub, maybe it's time to release a new version?

**Comment by tschaub** [comment]
Fix from @mrmlnc published in `mock-fs@4.11.0`.

Thanks for the fix and for the nudge, @mrmlnc.
